## Failing test suites <!--46088647575-->
Commit: f21f4a24e9e4939e4430e246640046640da71045 | [About building and testing Next.js](https://github.com/vercel/next.js/blob/canary/contributing.md)

`pnpm test-start test/e2e/prerender-native-module.test.ts`
- prerender native module > should render index correctly ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22prerender+native+module+should+render+index+correctly%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- prerender native module > should render /blog/first correctly ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22prerender+native+module+should+render+%2Fblog%2Ffirst+correctly%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- prerender native module > should render /blog/second correctly ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22prerender+native+module+should+render+%2Fblog%2Fsecond+correctly%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- prerender native module > should output traces ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22prerender+native+module+should+output+traces%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))

<details>
<summary>Expand output</summary>

  ● prerender native module › should render index correctly

    next build failed with code/signal 1

      90 |           if (code || signal)
      91 |             reject(
    > 92 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      93 |             )
      94 |           else resolve()
      95 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:92:15)

  ● prerender native module › should render /blog/first correctly

    next build failed with code/signal 1

      90 |           if (code || signal)
      91 |             reject(
    > 92 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      93 |             )
      94 |           else resolve()
      95 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:92:15)

  ● prerender native module › should render /blog/second correctly

    next build failed with code/signal 1

      90 |           if (code || signal)
      91 |             reject(
    > 92 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      93 |             )
      94 |           else resolve()
      95 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:92:15)

  ● prerender native module › should output traces

    next build failed with code/signal 1

      90 |           if (code || signal)
      91 |             reject(
    > 92 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      93 |             )
      94 |           else resolve()
      95 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:92:15)


  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'destroy')

      23 |     })
      24 |   })
    > 25 |   afterAll(() => next.destroy())
         |                       ^
      26 |
      27 |   it('should render index correctly', async () => {
      28 |     const browser = await webdriver(next.url, '/')

      at Object.destroy (e2e/prerender-native-module.test.ts:25:23)

</details>

`pnpm test-start test/e2e/app-dir/turbopack-reports/turbopack-reports.test.ts`
- turbopack-reports > should render page importing sqlite3 ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22turbopack-reports+should+render+page+importing+sqlite3%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))

<details>
<summary>Expand output</summary>

  ● turbopack-reports › should render page importing sqlite3

    next build failed with code/signal 1

      90 |           if (code || signal)
      91 |             reject(
    > 92 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      93 |             )
      94 |           else resolve()
      95 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:92:15)

</details>

`pnpm test-start test/production/build-spinners/index.test.ts`
- build-spinners > should handle build spinners correctly app dir - basic ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22build-spinners+should+handle+build+spinners+correctly+app+dir+-+basic%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- build-spinners > should handle build spinners correctly app dir - (compile workers) ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22build-spinners+should+handle+build+spinners+correctly+app+dir+-+%28compile+workers%29%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- build-spinners > should handle build spinners correctly page dir ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22build-spinners+should+handle+build+spinners+correctly+page+dir%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- build-spinners > should handle build spinners correctly page dir (compile workers) ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22build-spinners+should+handle+build+spinners+correctly+page+dir+%28compile+workers%29%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))
- build-spinners > should handle build spinners correctly app and pages ([DD](https://app.datadoghq.com/ci/test/runs?query=%40git.repository.id%3A%22github.com%2Fvercel%2Fnext.js%22+%40git.commit.head_sha%3A%22f21f4a24e9e4939e4430e246640046640da71045%22+%40test.name%3A%22build-spinners+should+handle+build+spinners+correctly+app+and+pages%22+%40test.type%3A%22nextjs%22+%40test.status%3A%22fail%22))

<details>
<summary>Expand output</summary>

  ● build-spinners › should handle build spinners correctly app dir - basic

    Cannot find module '../build/Release/pty.node' from '../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js'

    Require stack:
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/index.js
      production/build-spinners/index.test.ts

      111 |     const nextBin = resolveFrom(appDir, 'next/dist/bin/next')
      112 |     const ptyPath = resolveFrom(appDir, 'node-pty')
    > 113 |     const pty = require(ptyPath)
          |                 ^
      114 |     const output = []
      115 |     const ptyProcess = pty.spawn(process.execPath, [nextBin, 'build'], {
      116 |       name: 'xterm-color',

      at Resolver._throwModNotFoundError (../node_modules/.pnpm/jest-resolve@29.7.0/node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/unixTerminal.ts:14:9)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/index.ts:14:18)
      at require (production/build-spinners/index.test.ts:113:17)

  ● build-spinners › should handle build spinners correctly app dir - (compile workers)

    Cannot find module '../build/Release/pty.node' from '../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js'

    Require stack:
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/index.js
      production/build-spinners/index.test.ts

      111 |     const nextBin = resolveFrom(appDir, 'next/dist/bin/next')
      112 |     const ptyPath = resolveFrom(appDir, 'node-pty')
    > 113 |     const pty = require(ptyPath)
          |                 ^
      114 |     const output = []
      115 |     const ptyProcess = pty.spawn(process.execPath, [nextBin, 'build'], {
      116 |       name: 'xterm-color',

      at Resolver._throwModNotFoundError (../node_modules/.pnpm/jest-resolve@29.7.0/node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/unixTerminal.ts:14:9)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/index.ts:14:18)
      at require (production/build-spinners/index.test.ts:113:17)

  ● build-spinners › should handle build spinners correctly page dir

    Cannot find module '../build/Release/pty.node' from '../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js'

    Require stack:
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/index.js
      production/build-spinners/index.test.ts

      111 |     const nextBin = resolveFrom(appDir, 'next/dist/bin/next')
      112 |     const ptyPath = resolveFrom(appDir, 'node-pty')
    > 113 |     const pty = require(ptyPath)
          |                 ^
      114 |     const output = []
      115 |     const ptyProcess = pty.spawn(process.execPath, [nextBin, 'build'], {
      116 |       name: 'xterm-color',

      at Resolver._throwModNotFoundError (../node_modules/.pnpm/jest-resolve@29.7.0/node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/unixTerminal.ts:14:9)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/index.ts:14:18)
      at require (production/build-spinners/index.test.ts:113:17)

  ● build-spinners › should handle build spinners correctly page dir (compile workers)

    Cannot find module '../build/Release/pty.node' from '../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js'

    Require stack:
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/index.js
      production/build-spinners/index.test.ts

      111 |     const nextBin = resolveFrom(appDir, 'next/dist/bin/next')
      112 |     const ptyPath = resolveFrom(appDir, 'node-pty')
    > 113 |     const pty = require(ptyPath)
          |                 ^
      114 |     const output = []
      115 |     const ptyProcess = pty.spawn(process.execPath, [nextBin, 'build'], {
      116 |       name: 'xterm-color',

      at Resolver._throwModNotFoundError (../node_modules/.pnpm/jest-resolve@29.7.0/node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/unixTerminal.ts:14:9)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/index.ts:14:18)
      at require (production/build-spinners/index.test.ts:113:17)

  ● build-spinners › should handle build spinners correctly app and pages

    Cannot find module '../build/Release/pty.node' from '../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js'

    Require stack:
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/unixTerminal.js
      /tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/lib/index.js
      production/build-spinners/index.test.ts

      111 |     const nextBin = resolveFrom(appDir, 'next/dist/bin/next')
      112 |     const ptyPath = resolveFrom(appDir, 'node-pty')
    > 113 |     const pty = require(ptyPath)
          |                 ^
      114 |     const output = []
      115 |     const ptyProcess = pty.spawn(process.execPath, [nextBin, 'build'], {
      116 |       name: 'xterm-color',

      at Resolver._throwModNotFoundError (../node_modules/.pnpm/jest-resolve@29.7.0/node_modules/jest-resolve/build/resolver.js:427:11)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/unixTerminal.ts:14:9)
      at Object.<anonymous> (../../../../../../tmp/next-install-73f42f61462d94694668ee7436dd6edf1df57fa3f93402d6b9c746b3b26e2ca2/node_modules/.pnpm/node-pty@0.10.1/node_modules/node-pty/src/index.ts:14:18)
      at require (production/build-spinners/index.test.ts:113:17)

</details>

--- a/packages/next/src/client/flight-data-helpers.ts
+++ b/packages/next/src/client/flight-data-helpers.ts
@@ -30,13 +30,8 @@ export function prepareFlightRouterStateForRequest(
 function stripClientOnlyDataFromFlightRouterState(
   flightRouterState: FlightRouterState
 ): FlightRouterState {
-  const [
-    segment,
-    parallelRoutes,
-    _url, // Intentionally unused - URLs are client-only
-    refreshMarker,
-    isRootLayout,
-  ] = flightRouterState
+  const [segment, parallelRoutes, , refreshMarker, isRootLayout] =
+    flightRouterState
 
   // __PAGE__ segments are always fetched from the server, so there's
   // no need to send them up

